src/ssa/malloc_ssa.cpp

```diff
@@ -163,23 +163,26 @@ exprt malloc_ssa(
       if(tmp_type.is_not_nil())
       {
         // Did the size get multiplied?
-        mp_integer elem_size=pointer_offset_size(tmp_type, ns);
-        mp_integer alloc_size;
-        if(elem_size<0 || to_integer(size, alloc_size))
+        if(auto elem_size_ptr=pointer_offset_size(tmp_type, ns))
```

Comment:
Oh, that's one thing I was also hesitant about, you are right that `ptr` suffix doesn't make sense, the problem is finding a better alternative... Using `elem_size` and then using `*` or `value()` in calculations makes the code IMO harder to read (e.g. `if(elements**elem_size==alloc_size)` so I'd prefer dereferencing the optional value and saving it to a variable to keep the readability but the optional value and `mp_integer` must be named differently due to name clashes. Perhaps using `value()` is the way to go, rather than `*`

Reply:
Often we call it `maybe_elem_size` to indicate that it might not have a value.

Reply:
I like this suggestion, changed according to it.